### DISCUSSION TITLE
Thermal hydraulics: pressure fluctuations at small dt

### DISCUSSION THREAD

**Discussion by jvwilliams23** [idaholab/moose#25718] [Q&A Modules: General] [answered]
I have noticed some behaviour which I would say is non-physical in the thermal hydraulics module. At first I thought it was my own case configuration, but I have iteratively simplified the case all the way back to the tutorials ([recent tutorial committed](https://github.com/idaholab/moose/blob/cc2664f41c1e163e7761e11735fa637e12c282b9/modules/thermal_hydraulics/tutorials/single_phase_flow/01_flow_channel.i) by @licharlot).

**Differences from tutorial:** I reduced the outlet pressure to atmospheric pressure (`1e5 Pa`), and changed the fluid properties to water (`SimpleFluidProperties`). I also changed the initial velocity to `0 m/s`.

When `dt=10` as in the tutorial, the solve converges to steady state on the first step and the pressure drop is positive (pressure at inlet slightly higher than outlet). But when I set `dt=1e-4`, the flow is transient and the pressure drop fluctuates between positive and negative, then eventually converges to a steady-state solution with a **negative pressure drop**. 

My question is, am I misinterpreting the physics here? Or is there some numerical issue? Note that this also appears when I set `reversible=false` for the inlet BC. I thought it may be a floating point issue, since the flowrate is quite low, but also when I use a higher massflow rate (`m_dot_in = 1.0`), I get negative pressure-drop at the start and then it converges to a sensible solution with a positive pressure-drop.

```
T_in = 300. # K
m_dot_in = 1e-2 # kg/s
press = 1e5 # Pa

[GlobalParams]
  initial_p = ${press}
  initial_vel = 0.0
  initial_T = ${T_in}
  gravity_vector = '0 0 0'

  rdg_slope_reconstruction = minmod
  scaling_factor_1phase = '1 1e-2 1e-4'
  closures = thm_closures
  fp = water
[]

[FluidProperties]
  [water]
    type=SimpleFluidProperties
  []
[]

[Closures]
  [thm_closures]
    type = Closures1PhaseTHM
  []
[]

[Components]
  [inlet]
    type = InletMassFlowRateTemperature1Phase
    input = 'core_chan:in'
    m_dot = ${m_dot_in}
    T = ${T_in}
  []

  [core_chan]
    type = FlowChannel1Phase
    position = '0 0 0'
    orientation = '0 0 1'
    length = 1
    n_elems = 25
    A = 7.2548e-3
    D_h = 7.0636e-2
  []

  [outlet]
    type = Outlet1Phase
    input = 'core_chan:out'
    p = ${press}
  []
[]

[Postprocessors]
  [core_p_in]
    type = SideAverageValue
    boundary = core_chan:in
    variable = p
  []

  [core_p_out]
    type = SideAverageValue
    boundary = core_chan:out
    variable = p
  []

  [core_delta_p]
    type = ParsedPostprocessor
    pp_names = 'core_p_in core_p_out'
    function = 'core_p_in - core_p_out'
  []
[]

[Preconditioning]
  [pc]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  line_search = basic

  end_time="1000"
  dt="1e-04"
  # dt="10"
  dtmin="1e-12"

  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'

  nl_rel_tol = 1e-8
  nl_abs_tol = 1e-8
  nl_max_its = 25
  steady_state_detection="true"

[]

[Outputs]
  vtk=true
  csv=true
  print_linear_residuals = false
[]


```

**Comment by licharlot** [comment, marked as answer]
Hello, 

I removed `steady_state_detection=true`. If turned on, as is, it stops the simulation before the actual steady state. If you let it run, it'll converge to steady state, you'll see the pressure drop is not changing anymore, and it has a positive value, which is close to the one you get with dt=10 .(9.45e-3 Pa vs 9.047e-3 Pa).  It is expected that setting a very small dt, you'd observe a transient. The fluid is initially at rest and at t=0, a mass flow rate is imposed. Because THM is solving for compressible flow, it is solving the wave propagation in the domain and you get these negative pressure drops.
